pkgs/development/libraries/libxsmm/default.nix

```diff
@@ -35,10 +35,23 @@ stdenv.mkDerivation rec {
     "STATIC=${static}"
   ];
 
-  postInstall = ''
+  postInstall =
+  let rpath = lib.makeLibraryPath [
+        stdenv.cc.libc
+        stdenv.cc.cc.lib
+        gfortran.cc.lib
+        "$out"
+      ];
+  in ''
     mkdir -p $dev/lib/pkgconfig
     mv $out/lib/*.pc $dev/lib/pkgconfig
 
+    # Set rpath for all binaries and shared libraries explicitly.
+    # Otherwise the rpath contains /build/source/lib paths
+    for F in $out/bin/* $out/lib/*.so*; do
+      patchelf --set-rpath ${rpath} $F
+    done
```

Comment:
```suggestion
```
Completely re-writing the rpath is a pretty rough method.

Patching the makefiles is maybe a bit less intrusive?
This should do the job:
```
diff --git a/Makefile b/Makefile
index 15ee6f056..342502a07 100644
--- a/Makefile
+++ b/Makefile
@@ -875,7 +875,7 @@ endif
 .PHONY: generator
 generator: $(BINDIR)/libxsmm_gemm_generator
 $(BINDIR)/libxsmm_gemm_generator: $(BINDIR)/.make $(OBJFILES_GEN_GEMM_BIN) $(OUTDIR)/libxsmmgen.$(LIBEXT)
-       $(LD) -o $@ $(OBJFILES_GEN_GEMM_BIN) $(call abslib,$(OUTDIR)/libxsmmgen.$(ILIBEXT)) \
+       $(LD) -o $@ $(OBJFILES_GEN_GEMM_BIN) -Wl,--rpath $(PREFIX)/lib lib/libxsmmgen.$(ILIBEXT) \
                $(call cleanld,$(NOBLAS_LDFLAGS) $(NOBLAS_CLDFLAGS))
 
 ifneq (,$(strip $(LIBJITPROFILING)))
diff --git a/Makefile.inc b/Makefile.inc
index 424d7b2e4..87934fee0 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -918,7 +918,7 @@ ifneq (Darwin,$(UNAME))
   XLNKVERBOSE := --verbose
   linkopt = $(if $1,$(XLNKOPT)$(if $2,$1=$(call quote,$2),$1))
   abslibrpath = $(strip $(if $(findstring .$(ILIBEXT),$1)$(wildcard $1/), \
-    $(call linkopt,--rpath,$(call qxdir,$(call qapath,$1)))))
+    $(call linkopt,--rpath,$(PREFIX)/lib)))
   XGROUP_BEGIN := $(call linkopt,--start-group)
   XGROUP_END := $(call linkopt,--end-group)
   ifneq (0,$(ASNEEDED))
```

Reply:
Had to adapt a little bit for release 1.17 and not master, but yes, this is much better.

Reply:
Yes that one works and finds all libraries. The patched line in `Makefile` does not exist on 1.17. But admittetdly, when playing around with that Git Repo, it behaved somewhat weird. Had problems checking out tags.